frontend/components/Header/Nav/Pillars/pillars.ts

```diff
@@ -0,0 +1,27 @@
+import { PillarColours, palette } from '@guardian/pasteup/palette';
+
+export const pillarNames: Pillar[] = [
+    'news',
+    'opinion',
+    'sport',
+    'culture',
+    'lifestyle',
+];
+
+export const pillarPalette: { [K in Pillar]: PillarColours } = {
+    news: palette.news,
+    opinion: palette.opinion,
+    sport: palette.sport,
+    culture: palette.culture,
+    lifestyle: palette.lifestyle,
+};
+
+export const pillarMap: <T>(
+    f: (name: Pillar) => T,
+) => { [K in Pillar]: T } = f => ({
+    news: f('news'),
+    opinion: f('opinion'),
+    sport: f('sport'),
+    culture: f('culture'),
+    lifestyle: f('lifestyle'),
+}); // We could do this with an Object.assign, but that would need weirdness in ts and is less readable.
```

Comment:
This is really difficult to follow. I would much rather disable the type checking that is causing a problem here and set [`suppressImplicitAnyIndexErrors`](https://github.com/Microsoft/TypeScript/issues/1232#issuecomment-64510362) to `true` in the [`tsconfig`](https://www.typescriptlang.org/docs/handbook/compiler-options.html).